### PR TITLE
Regen

### PR DESCRIPTION
**Pull request by GuillaumeGomez**
cc @sdroege @EPashkin

### CODE REVIEW

**Comment by sdroege** [comment]
:+1: 

Also you can remove 1.28 from CI and update to 1.31. Most of the ecosystem unfortunately moved to 1.31 already.

File: src/auto/attr_list.rs [hunk 1]
```diff
@@ -30,9 +31,20 @@ impl AttrList {
...
     }
 
-    //pub fn filter<P: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(&self, func: /*Unknown conversion*//*Unimplemented*/AttrFilterFunc, data: P) -> Option<AttrList> {
-    //    unsafe { TODO: call ffi::pango_attr_list_filter() }
-    //}
+    pub fn filter<P: FnMut(&Attribute) -> bool>(&self, func: P) -> Option<AttrList> {
+        let func_data: Option<P> = func.into();
+        unsafe extern "C" fn func_func<P: FnMut(&Attribute) -> bool>(attribute: *mut ffi::PangoAttribute, user_data: glib_ffi::gpointer) -> glib_ffi::gboolean {
+            let attribute = from_glib_borrow(attribute);
+            let callback: *mut P = user_data as *const _ as usize as *mut P;
```
Comment (sdroege) [inline src/auto/attr_list.rs]: This is also wrong. The `Option<_>` got lost

**Comment by EPashkin** [comment]
> Most of the ecosystem unfortunately moved to 1.31 already.

Why "unfortunately" ?

**Comment by sdroege** [comment]
> > Most of the ecosystem unfortunately moved to 1.31 already.
> 
> Why "unfortunately" ?

Because it makes life more difficult for e.g. enterprise Linux distros (which are currently all at 1.28), and generally for people who can't just update directly.

**Comment by sdroege** [comment]
:+1:

**Pull request merged**